src/zarr/v3/codecs/blosc.py

```diff
@@ -1,85 +1,160 @@
+    from typing import Dict, Optional
+    from typing_extensions import Self
+    from zarr.v3.common import JSON, ArraySpec, BytesLike, RuntimeConfiguration
 
 
-BloscShuffle = Literal["noshuffle", "shuffle", "bitshuffle"]
+class BloscShuffle(Enum):
+    noshuffle = "noshuffle"
+    shuffle = "shuffle"
+    bitshuffle = "bitshuffle"
+
+    @classmethod
+    def from_int(cls, num: int) -> Self:
+        blosc_shuffle_int_to_str = {
+            0: "noshuffle",
+            1: "shuffle",
+            2: "bitshuffle",
+        }
+        if num not in blosc_shuffle_int_to_str:
+            raise ValueError(f"Value must be between 0 and 2. Got {num}.")
+        return BloscShuffle[blosc_shuffle_int_to_str[num]]
+
+
+class BloscCname(Enum):
+    lz4 = "lz4"
+    lz4hc = "lz4hc"
+    blosclz = "blosclz"
+    zstd = "zstd"
+    snappy = "snappy"
+    zlib = "zlib"
+
 
 # See https://zarr.readthedocs.io/en/stable/tutorial.html#configuring-blosc
 numcodecs.blosc.use_threads = False
 
 
-@frozen
-class BloscCodecConfigurationMetadata:
-    typesize: int
-    cname: Literal["lz4", "lz4hc", "blosclz", "zstd", "snappy", "zlib"] = "zstd"
-    clevel: int = 5
-    shuffle: BloscShuffle = "noshuffle"
-    blocksize: int = 0
+def parse_typesize(data: JSON) -> int:
+    if isinstance(data, int):
+        if data > 0:
+            return data
+        else:
+            raise ValueError(
+                f"Value must be greater than 0. Got {data}, which is less or equal to 0."
+            )
+    raise TypeError(f"Value must be an int. Got {type(data)} instead.")
 
 
-blosc_shuffle_int_to_str: Dict[int, BloscShuffle] = {
-    0: "noshuffle",
-    1: "shuffle",
-    2: "bitshuffle",
-}
+# todo: real validation
+def parse_clevel(data: JSON) -> int:
+    if isinstance(data, int):
+        return data
+    raise TypeError(f"Value should be an int. Got {type(data)} instead.")
 
 
-@frozen
-class BloscCodecMetadata:
-    configuration: BloscCodecConfigurationMetadata
-    name: Literal["blosc"] = field(default="blosc", init=False)
+def parse_blocksize(data: JSON) -> int:
+    if isinstance(data, int):
+        return data
+    raise TypeError(f"Value should be an int. Got {type(data)} instead.")
 
 
-@frozen
+@dataclass(frozen=True)
 class BloscCodec(BytesBytesCodec):
-    configuration: BloscCodecConfigurationMetadata
     is_fixed_size = False
 
-    @classmethod
-    def from_metadata(cls, codec_metadata: CodecMetadata) -> BloscCodec:
-        assert isinstance(codec_metadata, BloscCodecMetadata)
-        return cls(configuration=codec_metadata.configuration)
+    typesize: int
+    cname: BloscCname = BloscCname.zstd
+    clevel: int = 5
+    shuffle: BloscShuffle = BloscShuffle.noshuffle
+    blocksize: int = 0
+
+    def __init__(
+        self,
+        *,
+        typesize=None,
+        cname=BloscCname.zstd,
+        clevel=5,
+        shuffle=None,
+        blocksize=0,
+    ) -> None:
```

Comment:
I wonder if we should add type information to the `__init__` signatures? My vscode seems to take these for intellisense/autocomplete.

Reply:
yeah, I think these should be annotated with the types of the attributes. I can add annotations when I have time, but if you beat me to it I won't complain :)

Reply:
resolved by [fb18e9d](https://github.com/zarr-developers/zarr-python/pull/1660/commits/fb18e9d6281004daf658db756df51406a780e82f) and [4c448dc](https://github.com/zarr-developers/zarr-python/pull/1660/commits/4c448dc3a84eed2cf39faea2011f82ec7034bcd7)